In [1]:
import pandas as pd
import numpy as np
import tensorflow as tf
from tensorflow import keras
from sklearn.model_selection import train_test_split

In [2]:
# test and train split
alz_file = pd.read_csv('12_months_joint.csv')
X = alz_file.drop(['cov'], axis = 1)
Y = alz_file['cov']
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.2, random_state=42)

In [26]:
# single layer perceptron implementation
model = keras.Sequential([keras.layers.Dense(1, input_dim=X_train.shape[1], activation='sigmoid')])
model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
model.fit(X_train, Y_train, epochs=250, batch_size=32, verbose=0)

loss, accuracy = model.evaluate(X_test, Y_test)
print(f'Accuracy: {accuracy}')

2/2 [==============================] - 0s 12ms/step - loss: 0.4856 - accuracy: 0.8475
Accuracy: 0.8474576473236084


Metric readings with different parameters.
1. Params: 250 iterations (epochs), activation = sigmoid, batch_size = 10.
   Result: loss: 0.4149 - accuracy: 0.8305
2. Params: 250 iterations (epochs), activation = relu, batch_size=10.
   Result: loss: 10.3414 - accuracy: 0.3220
3. Params: 100 iterations (epochs), activation = sigmoid, batch_size=32
   Result: 2.6599 - accuracy: 0.7627
4. Params: 500 iterations (epochs), activation = relu, batch_size=32
   Result: 4.9674 - accuracy: 0.6780
5. Params: 500 iterations (epochs), activation = sigmoid, batch_size=32
   Result: loss: 0.4856 - accuracy: 0.8475
